Yank/restraints.py

```diff
@@ -551,20 +698,295 @@ def _determineBondParameters(self):
+    20 kcal/(mol A**2) or 20 kcal/(mol rad**2) as in Ref [1].
+
+    Future iterations of this feature will introduce the ability to extract equilibrium parameters and spring constants
+    from a short simulation.
+
+    WARNING
+    -------
+    Symmetry corrections for symmetric ligands are not automatically applied.
+    See Ref [1] and [2] for more information on correcting for ligand symmetry.
+
+    References
+    ----------
+    [1] Boresch S, Tettinger F, Leitgeb M, Karplus M. J Phys Chem B. 107:9535, 2003.
+        http://dx.doi.org/10.1021/jp0217839
+    [2] Mobley DL, Chodera JD, and Dill KA. J Chem Phys 125:084902, 2006.
+        https://dx.doi.org/10.1063%2F1.2221683
+
+    """
+
+    def __init__(self, topology, state, system, positions, receptor_atoms, ligand_atoms):
+        """
+        Initialize a receptor-ligand restraint class.
+
+        Parameters
+        ----------
+        state : thermodynamics.ThermodynamicState
+            The thermodynamic state specifying temperature, pressure, etc. to which restraints are to be added
+        system : simtk.openmm.System
+            OpenMM representation of fully interacting system
+        positions : simtk.unit.Quantity of natoms x 3 with units compatible with nanometers
+            Reference positions to use for imposing restraints
+        receptor_atoms : list of int
+            A complete list of receptor atoms
+        ligand_atoms : list of int
+            A complete list of ligand atoms
+
+        """
+        super(Boresch, self).__init__(topology, state, system, positions, receptor_atoms, ligand_atoms)
+
+        # Select atoms to be used in restraint.
+        self.restraint_atoms = self._select_restraint_atoms(positions, receptor_atoms, ligand_atoms)
+
+        # Determine restraint parameters
+        self._determine_restraint_parameters(state, system, positions)
+
+    def _is_collinear(self, positions, atoms, THRESHOLD=0.9):
+        """Report whether any sequential vectors in a sequence of atoms are collinear to within a given dot product threshold.
+
+        Parameters
+        ----------
+        positions : simtk.unit.Quantity of natoms x 3 with units compatible with nanometers
+            Reference positions to use for imposing restraints
+        atoms : array-like, dtype:int, length:natoms
+            A seris of indices of atoms
+        THRESHOLD : float, optional, default=0.9
+            Atoms are not colinear if their sequential vector separation dot products are less than THRESHOLD
+
+        Returns
+        -------
+        result : bool
+            Returns True if any sequential pair of vectors is collinear; False otherwise.
+
+        """
+        result = False
+        for i in range(len(atoms)-2):
+            v1 = positions[atoms[i+1],:] - positions[atoms[i],:]
+            v2 = positions[atoms[i+2],:] - positions[atoms[i+1],:]
+            innerprod = np.dot(v1, v2)
```

Comment:
Do we need to make `v1` and `v2` unit vectors before taking the inner product? Otherwise the threshold is relative to the distances between atoms.